### PR TITLE
[Custom Fields] Code cleanup

### LINKED ISSUE #12994: [Custom Fields] Remove feature flag and cleanup the code

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #12994 

### Description
This PR removes the Custom Fields feature flag and the old unused code.

### Steps to reproduce
For products:
1. Open a product.
2. Tap on the Custom Fields row, or on Add more details then Custom Fields options.

For orders:
1. Open an order.
2. Tap on "Custom Fields ✏️"

### Testing information
Confirm the feature is not impacted, just opening the screens should be enough.

### The tests that have been performed
^

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/customfields/CustomOrderFieldsScreen.kt [hunk 1]
```diff
@@ -1,202 +0,0 @@
...
-            start = 0,
-            end = value.length - 1
-        )
-        toAnnotatedString()
-    }
-
-    Text(
-        text = text,
-        style = MaterialTheme.typography.body2.copy(
-            color = colorResource(R.color.color_text_link)
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/customfields/CustomOrderFieldsScreen.kt:156]: It seems `R.color.color_text_link` is also unused now and we can remove it from `colors.xml`.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/customfields/CustomOrderFieldsScreen.kt [hunk 2]
```diff
@@ -1,202 +0,0 @@
...
-import com.woocommerce.android.ui.orders.details.customfields.CustomOrderFieldsHelper.CustomOrderFieldType
-import org.wordpress.android.fluxc.model.metadata.WCMetaData
-
-private var clickListener: CustomOrderFieldClickListener? = null
-
-@Composable
-fun CustomOrderFieldsScreen(viewModel: OrderDetailViewModel, listener: CustomOrderFieldClickListener? = null) {
-    clickListener = listener
-    CustomFieldsScreen(
-        viewModel.getOrderMetadata(),
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/customfields/CustomOrderFieldsScreen.kt:50]: `getOrderMetadata()` is now unused in `OrderDetailViewModel` and we can remove it.

**Review by irfano — APPROVED** [review]
LGTM! 👍🏻 I approved it, but I added comments suggesting that more code could be removed.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/customfields/CustomOrderFieldsScreen.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/customfields/CustomOrderFieldsScreen.kt:156]: Nice catch, removed in [7e21e1b](https://github.com/woocommerce/woocommerce-android/pull/12995/commits/7e21e1b9a6cf191d6edf413272e2da32e3817a8e)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/customfields/CustomOrderFieldsScreen.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/customfields/CustomOrderFieldsScreen.kt:50]: Removed in [7e21e1b](https://github.com/woocommerce/woocommerce-android/pull/12995/commits/7e21e1b9a6cf191d6edf413272e2da32e3817a8e), thanks

**Pull request merged**